kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -92,18 +196,68 @@ object Async:
 
         extension [E, A](self: Fiber[E, A])
 
+            /** Gets the result of the Fiber.
+              *
+              * @return
+              *   The result of the Fiber
+              */
             def get(using reduce: Reducible[Abort[E]], frame: Frame): A < (reduce.SReduced & Async) =
                 Async.get(self)
 
+            /** Uses the result of the Fiber to compute a new value.
+              *
+              * @param f
+              *   The function to apply to the Fiber's result
+              * @return
+              *   The result of applying the function to the Fiber's result
+              */
             def use[B, S](f: A => B < S)(using reduce: Reducible[Abort[E]], frame: Frame): B < (reduce.SReduced & Async & S) =
                 Async.use(self)(f)
 
-            def getResult(using Frame): Result[E, A] < Async                            = Async.getResult(self)
+            /** Gets the result of the Fiber as a Result.
+              *
+              * @return
+              *   The Result of the Fiber
+              */
+            def getResult(using Frame): Result[E, A] < Async = Async.getResult(self)
+
+            /** Uses the Result of the Fiber to compute a new value.
+              *
+              * @param f
+              *   The function to apply to the Fiber's Result
+              * @return
+              *   The result of applying the function to the Fiber's Result
+              */
             def useResult[B, S](f: Result[E, A] => B < S)(using Frame): B < (Async & S) = Async.useResult(self)(f)
-            def isDone(using Frame): Boolean < IO                                       = IO(self.isDone())
-            def onComplete(f: Result[E, A] => Unit < IO)(using Frame): Unit < IO        = IO(self.onComplete(r => IO.run(f(r)).eval))
-            def block(timeout: Duration)(using Frame): Result[E | Timeout, A] < IO      = IO(self.block(deadline(timeout)))
 
+            /** Checks if the Fiber is done.
+              *
+              * @return
+              *   Whether the Fiber is done
+              */
+            def isDone(using Frame): Boolean < IO = IO(self.isDone())
```

Comment:
Should we rename this to `done`?

Reply:
ah, that's an oversight. The refactoring that removed the `Done` fiber type should have renamed this as well. I'll follow up.